A29-xds-tls-security.md

```diff
@@ -379,58 +389,46 @@ on the client side (inside [`UpstreamTlsContext`][UTC]) is ignored if present.
 The processing and parsing of `CommonTlsContext` inside any particular CDS or LDS response does
 not take into account whether Xds credentials are in effect for the respective channel or server.
 
-[`combined_validation_context`][CVC] is validated in any CDS/LDS update as follows:
-The field [match_subject_alt_names][] inside [default_validation_context][] is accepted only on
-the client side i.e. inside `UpstreamTlsContext` and is processed as described [above][server-authz].
-If any of the other fields (listed below) are present in [default_validation_context][], the update
-is NACKed because ignoring these fields compromises security:
+A `CertificateValidationContext` (either the [`validation_context`][validation_context] field or
+the [`default_validation_context`][default_validation_context] field inside of the
+[`combined_validation_context`][CVC]) is validated in any CDS/LDS update as follows:
+The field [match_subject_alt_names][] is accepted only on the client side i.e. inside
```

Comment:
The wording here implies that if `match_subject_alt_names` is specified on the server side, we NACK.  But I think we should actually just ignore it in that case, right?

Reply:
You are right: we should clearly specify whether we NACK or just ignore. In terms of what we should do: I am okay to ignore it but it might be worthwhile to see what Envoy does. If I can figure it out, I'll update this thread

Reply:
I didn't spend too much time on analyzing the Envoy code but I didn't see Envoy NACKIng this on the server side. So I just modified the wording to say "we ignore it on the server side"

Reply:
I think we should either support this field on server-side or NACK. Following the principal of "NACKing things that are unsupported and would reduce security," this would certainly reduce security.

I agree it may be rarely used, but it can be used very similarly to RBAC (especially L4 RBAC).

> but I didn't see Envoy NACKIng this on the server side

I'd assume they actually support it on server-side. I see no reason for them to special-case that code.